helios/loan-escrow.js

```diff
@@ -0,0 +1,44 @@
+spending loan
+
+struct Datum {
+    lender: PubKeyHash
+    borrower: PubKeyHash
+    deadline: Time
+}
+
+enum Redeemer {
+    WithdrawADA
+    WithdrawCol
+}
+
+func main(datum: Datum, redeemer: Redeemer, context: ScriptContext) -> Bool {
+    tx: Tx = context.tx;
+    now: Time = tx.time_range.start;
+
+    redeemer.switch {
```

Comment:
Lender Withdraws || Borrower withdraws

Lender can withdraw any time while nft token isn't in the smart contract address.
Borrower can withdraw if and only if a token is in contract address;

Reply:
- [x] ~~borrower claims~~ https://github.com/aleeusgr/potential-robot/issues/85
- [x] borrower withdraws ada
- [ ] timeout
- [x] lender withdraws Collateral
- [x] alice locks, alice can unlock, anyone can unlock if validator has a utxo with the nft.